metafix/src/main/java/org/metafacture/metafix/Metafix.java

```diff
@@ -171,7 +171,13 @@ public void putMacro(final String name, final RecordTransformer macro) {
     }
 
     public RecordTransformer getMacro(final String name) {
-        return macros.get(name);
+        final RecordTransformer macro = macros.get(name);
+
+        if (macro != null) {
+            macro.setParentExceptionMessage(recordTransformer);
```

Comment:
Minor: I find the `set...` naming where the argument is not the thing that is set slightly confusing. Maybe `setParentExceptionMessageFrom`? Or instead of a helper method, the single-line implementation could go right here.

Reply:
I see what you mean. Changed in 36cd3da.

> Or instead of a helper method, the single-line implementation could go right here.

But then we would have to expose `currentMessageSupplier`, which I'd rather not.